release/create_release_archives.sh

```diff
@@ -107,8 +107,10 @@ function create_windows_archive() {
   rm "${istioctl_path}"
 }
 
-pushd "${OUTPUT_PATH}"
-${CP} istio.VERSION LICENSE README.md "${COMMON_FILES_DIR}"/
+# Not sure why it was used before - but the files are in source, not out
+# pushd "${OUTPUT_PATH}"
```

Comment:
Sounds good - but right now it doesn't seem to be working, and since updateVersion is mostly going away (most logic is in helm template) I don't think 'scribble the source while building the release' is a huge issue.

I think it's a larger issue that it's generating stuff in the source tree for normal development, where it affects a lot of people - but proper fix is in fixing updateVersion to generate in out, not in release script to work around.

Reply:
let's not break the official release process though

it makes sense it needs a "clean" git tree (for instance to test for cleanness)

is there a way to run a PR through the daily build process ?

the last one https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/
is still 3/25 and I don't want that we continue breaking it - we need those builds for perf testing and for sanity checks and to follow our daily build process

cc on call @cmluciano

Reply:
> is there a way to run a PR through the daily build process ?

The closest way to do a test build is to commit changes to a forked repo of istio (e.g., "master4" branch of "mattdelco/istio"), create a repo-tool manifest file that points to that fork (e.g., I use a "mattdelco/imanifest" repo where I've basically got an xml file for each branch I use [which is partly why I tend to reuse branches with boring names] like  https://github.com/mattdelco/manifest/blob/master/master4.xml ) and then use my gcloud acct to request a Cloud Builder build that stores the results in GCR/GCS locations that belong to that account), e.g.:

<pre>P_ID=${USER}-experimental
SVC_ACCT="cloudbuild@${P_ID}.iam.gserviceaccount.com"

MFEST_URL=https://github.com/mattdelco/manifest
MFEST_FILE="master4.xml"
MFEST_COMMIT="master"
VERSION="0.7.0-20180322"

GCR_DEST="${P_ID}/prerelease"
GCS_DEST="delco-prerelease/prerelease/${VERSION}"

./release/start_gcb_build.sh -w -p "${P_ID}" -a "${SVC_ACCT}" -r "${GCR_DEST}" -s "${GCS_DEST}" \
  -v "${VERSION}" -u "${MFEST_URL}" -t "${MFEST_COMMIT}" -m "${MFEST_FILE}"
</pre>

If you've got a Linux machine with docker, fpm, and gcloud/gsutil installed then in theory you can also run the same steps of the build on your local system, e.g.:

<pre>SCRATCH_DIR=/tmp/output
GCLOUD_ACCT=${USER}-experimental
VER_STRING=0.7.0-20180328
GCS_PATH=${GCLOUD_ACCT}/testbuild/${VER_STRING}
GCR_PATH=${GCLOUD_ACCT}

mkdir -p ${SCRATCH_DIR}
./release/cloud_builder.sh -p ${GCS_PATH} -q ${GCR_PATH} -t ${VER_STRING} -o ${SCRATCH_DIR}
./release/store_artifacts.sh -p ${GCS_PATH} -q ${GCR_PATH} -v ${VER_STRING} -o {SCRATCH_DIR}
</pre>

If you don't want to bother with Google infra then it's probably sufficient to just run cloud_builder.sh (the script takes GCS/GCR paths as input but I think they're only used by that script [via updateVersion.sh] to bake the paths into the product.